### DISCUSSION TITLE
Is there a way to expand CSS variables?

### DISCUSSION THREAD

**Discussion by arg** [cssnano/cssnano#1516] [Q&A]
Our website uses Tailwind CSS which implements a lot of CSS variables (e.g. `--tw-ring-color`, `--tw-ring-offset-color` and many more) These variables actually make the output CSS file larger as they are called like `var(<variable_name>)` in many places. If instead we could just put the actual values everywhere, our output CSS would be significantly smaller. Just to compare:

| Variable                 | Value     | How it's used in CSS file     | How many characters it takes | How many characters it would take without CSS variable |
|--------------------------|-----------|-------------------------------|------------------------------|--------------------------------------------------------|
| `--tw-ring-color`        | `#333`    | `var(--tw-ring-color)`        | 20                           | 4                                                      |
| `--tw-ring-offset-color` | `#434343` | `var(--tw-ring-offset-color)` | 27                           | 7                                                      |

I couldn't find any related configuration options in cssnano's documentation but maybe there is some way to configure it to replace all `var(...)` occurrences with the actual values during minification?

**Comment by wmertens** [comment]
Not quite what you want but there's already this https://github.com/navanshu/postcss-variable-compress

**Comment by wmertens** [comment]
besides, it's hard to determine which variables are actually used variably. If you can guarantee that all JS and CSS is built at the same time, you could at least remove the vars that are not in any CSS, and you could inline all the vars with only a single declaration. If JS contains the variable name, you can't remove or inline it.

**Comment by arg** [comment]
@wmertens thanks for the response, we're testing `postcss-preset-env` (with [`postcss-custom-properties`](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-custom-properties) plugin) and https://github.com/MadLittleMods/postcss-css-variables now.

**Reply by wmertens** [reply]
Cool! Please report back. That latter plugin seems great if you can provide it with a list of variables that are mentioned within JS, then it should clear out all the rest.
